File: core/src/main/java/com/nfthub/core/entity/Artist.java
package com.nfthub.core.entity;

import lombok.Getter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table
@Getter
public class Artist {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private Category category;

    @OneToMany(mappedBy = "artist", cascade = {CascadeType.ALL})
    private List<ArtistImage> artistImages = new ArrayList<>();

    @OneToMany(mappedBy = "artist", cascade = {CascadeType.ALL})
    private List<ArtistTag> artistTags = new ArrayList<>();

}

File: src/main/kotlin/com/nfthub/api/entity/Artist.kt
package com.nfthub.api.entity

import com.nfthub.api.EMPTY_STRING
import javax.persistence.*

@Entity
@Table
class Artist(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    var id: Long = 0,

    var view: Long = 0,

    @ManyToOne(fetch = FetchType.LAZY, optional = true)
    var category: Category? = null,

    @OneToMany(mappedBy = "artist", orphanRemoval = true, cascade = [CascadeType.ALL])
    var artistKeywords: List<ArtistKeyword> = emptyList(),

    @OneToMany(mappedBy = "artist", orphanRemoval = true, cascade = [CascadeType.ALL])
    var artistImages: List<ArtistImage> = emptyList(),

    @OneToMany(mappedBy = "artist", orphanRemoval = true, cascade = [CascadeType.ALL])
    var artistArt: List<Art> = emptyList()
) : BaseTimeEntity()

@Entity
@Table
class ArtistKeyword(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    var id: Long = 0,

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    var artist: Artist = Artist()
)

@Entity
@Table
class ArtistImage(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false)
    var id: Long = 0,
    @Column(columnDefinition = "text", length = 500)
    var url: String = EMPTY_STRING,
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    var artist: Artist = Artist(),
    var isMain: Boolean = false,
)